tests/bluetooth/classic/gap_c/pytest/test_gap_c.py

```diff
@@ -0,0 +1,478 @@
+        device = Device.with_hci(
+            'Bumble',
+            Address('F0:F1:F2:F3:F4:F5'),
+            hci_transport.source,
+            hci_transport.sink,
+        )
+        device.classic_enabled = True
+        device.le_enabled = False
+        device.listener = DiscoveryListener()
+
+        with open(f"bumble_hci_{sys._getframe().f_code.co_name}.log", "wb") as snoop_file:
+            device.host.snooper = BtSnooper(snoop_file)
+            await device.power_on()
+            dongle_address = str(device.public_address).replace(r'/P', '')
+            # Start of Initial Condition
+            await device.set_discoverable(True)  # Set peripheral as discoverable
+            await device.set_connectable(True)  # Set peripheral as connectable
+            await device.send_command(HCI_Write_Page_Timeout_Command(page_timeout=0xFFFF))
+            shell.exec_command("bt disconnect")
+            # End of Initial Condition
+
+            # Test Start
+            logger.info("Step 1: DUT initiates general inquiry")
+            # Use limited inquiry as the control group
+            await send_cmd_to_iut(shell, dut, "br discovery on", dongle_address)
+
+            logger.info("Step 2: Tester responds to the inquiry")
+            logger.info("This is a passive step and it always succeed.")
+
+            logger.info("Step 3: DUT sends connect request to tester")
+            await send_cmd_to_iut(shell, dut, f"br connect {dongle_address}", "Connected")
+
+            logger.info(
+                "Step 4: Tester accepts the connection request and connected event is received"
+            )
+            logger.info("This is a passive step and it always succeed.")
+
+            logger.info("Step 5: Tester initiates disconnection")
+            connection = device.find_connection_by_bd_addr(Address(dut_address))
+            assert connection is not None, "No connection found with the DUT"
+            await connection.disconnect()
+
+            logger.info("Step 6: Connection is terminated")
+            found, _ = await _wait_for_shell_response(dut, "Disconnected")
+            assert found, "Disconnection event not received"
+
+
+async def tc_gap_c_3(hci_port, shell, dut, address) -> None:
+    case_name = 'GAP-C-3: General Inquiry followed by Rejected Connection Request'
+    logger.info(f'<<< Start {case_name} ...')
+
+    async with await open_transport_or_link(hci_port) as hci_transport:
+        # init Dongle bluetooth
+        device = Device.with_hci(
+            'Bumble',
+            Address('F0:F1:F2:F3:F4:F5'),
+            hci_transport.source,
+            hci_transport.sink,
+        )
+        device.classic_enabled = True
+        device.le_enabled = False
+        device.listener = DiscoveryListener()
+
+        with open(f"bumble_hci_{sys._getframe().f_code.co_name}.log", "wb") as snoop_file:
+            device.host.snooper = BtSnooper(snoop_file)
+            await device.power_on()
```

Comment:
```suggestion
            await device_power_on()
```

```
async def device_power_on(device) -> None:
    while True:
        try:
            await device.power_on()
            break
        except Exception:
            continue
```

The changes is used to make sure the test case can work in Linux. It needs to be applied for all test cases.